pkg/image/origins/main.go

```diff
@@ -0,0 +1,284 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"github.com/coreos/go-semver/semver"
+	"github.com/pkg/errors"
+	kd "github.com/rancher/rancher/pkg/controllers/management/kontainerdrivermetadata"
+	img "github.com/rancher/rancher/pkg/image"
+	ext "github.com/rancher/rancher/pkg/image/external"
+	"github.com/rancher/rancher/pkg/settings"
+	"github.com/rancher/rke/types/kdm"
+)
+
+// This file attempts to create an updated mapping of images used in rancher and their source code origin.
+// It does so on a best-effort basis and prints the resulting map to the terminal. Rancher images which share a name
+// with its base repository (such as gke-operator, which comes from github.com/rancher/gke-operator) will be
+// automatically resolved. In the case where images are detected but cannot be automatically resolved, a warning
+// will be printed to the console indicating such images, and the value of the image in the map will be 'unknown'.
+//
+// This tool can be run with the command `dapper check-origins`.
+
+func main() {
+	if err := inner(os.Args[1], os.Args[2], os.Args[3:]); err != nil {
+		panic(err)
+	}
+}
+
+const imageNotFound = "image not found"
+
+func inner(systemChartsPath, chartsPath string, imagesFromArgs []string) error {
```

Comment:
Most of this function is the same logic as the function `run` in `pkg/image/export/main.go` 
How about extracting the common parts to make a helper function?

Reply:
`utilities.go` now handles the processing of the images and image sources within `GatherTargetImagesAndSources`, which is used by both `img/export/main.go` and `img/origins/main.go`.